node/pkg/fetcher/localaggregator.go

```diff
@@ -114,21 +116,50 @@ func (c *LocalAggregator) processVolumeWeightedFeeds(ctx context.Context, feeds
 }
 
 func filterOutliers(feeds []*FeedData) ([]*FeedData, error) {
-	median, err := calculateMedian(feeds)
+	if len(feeds) < 5 {
+		return feeds, nil
+	}
+
+	data := make([]float64, len(feeds))
+	for i, feed := range feeds {
+		data[i] = feed.Value
+	}
+
+	outliers, err := stats.QuartileOutliers(data)
 	if err != nil {
 		return nil, err
 	}
 
-	var filteredFeeds []*FeedData
-	for _, feed := range feeds {
-		price := feed.Value
-		priceDifference := math.Abs((median - price) / median)
-		if priceDifference > OutlierThreshold {
-			continue
+	maxOutliersToRemove := int(float64(len(feeds)) * MaxOutlierRemovalRatio)
+	totalOutliers := len(outliers.Extreme) + len(outliers.Mild)
+
+	if totalOutliers > maxOutliersToRemove {
```

Comment:
🤔 nice point, but I'm not sure if mild extrems should be taken into account and
the logic should take branch separately, 

if you see the logic inside, it reorders outliers based on its difference from median. and maximum 25% of the data can be filtered out. so it will remove only extreme outliers first in order

ex)
total num of data:20
extreme 3 + mild 3 => will remove 3 extreme and 2 mild
extreme 3 => will remove 3 extreme
extreme 1 + mild 5 => will remove 1 extreme 4 mild

Reply:
awh wait but I'm not sure if checking the difference from median assures the order from extreme -> mild outliers, will check that out too

Reply:
Found out that it doesn't necessarily order from extreme -> mild order
so I'll make some change with the logic